### PR TITLE
Enforce a line length limit.

### PR DESCRIPTION
**Pull request by RoosterDragon**
Prevent lines longer than 180 chars (tabs count as 4) via a SyleCopPlus rule.

It is open for discussion whether we want to enforce such a rule at all, and what values to use. I picked the current value as an arbitrary number to prevent crazy long lines but which didn't require me to fix a billion files.

### CODE REVIEW

**Comment by Phrohdoh** [comment]
I'm okay with this and feel 180 is a reasonable length. :+1:

File: Settings.StyleCop [hunk 1]
```diff
@@ -394,8 +394,15 @@
             <BooleanProperty Name="Enabled">False</BooleanProperty>
           </RuleSettings>
         </Rule>
+		<Rule Name="CodeLineMustNotBeLongerThan">
```
Comment (ScottNZ) [inline Settings.StyleCop]: The indentation method here is inconsistent (tabs instead of spaces)

**Comment by penev92** [comment]
180 may be a bit much for some people.

**Comment by RoosterDragon** [comment]
Yeah, I'd like a lower value too - but there's a lot of files that would need tweaking to get there.

**Comment by pchote** [comment]
This seems reasonable, so no objection from me. A future pr could possibly lower the limit a bit more, but I suspect that the verboseness of c# would make it a struggle to get below 120 chars.

**Comment by penev92** [comment]
Down to 130 would be enough. ;)

**Comment by cjshmyr** [comment]
I'm good with the initial changes here before lowering it.

**Comment by Phrohdoh** [comment]
Just so we're clear my :+1: is for 180.

File: OpenRA.Mods.Common/Traits/Air/Plane.cs [hunk 2]
```diff
@@ -129,12 +129,14 @@ public void ResolveOrder(Actor self, Order order)
...
-		public Activity MoveWithinRange(Target target, WRange minRange, WRange maxRange) { return Util.SequenceActivities(new Fly(self, target, minRange, maxRange), new FlyCircle()); }
+		public Activity MoveWithinRange(Target target, WRange minRange, WRange maxRange)
+		{ return Util.SequenceActivities(new Fly(self, target, minRange, maxRange), new FlyCircle()); }
 		public Activity MoveFollow(Actor self, Target target, WRange minRange, WRange maxRange) { return new FlyFollow(self, target, minRange, maxRange); }
 		public CPos NearestMoveableCell(CPos cell) { return cell; }
 
 		public Activity MoveIntoWorld(Actor self, CPos cell, SubCell subCell = SubCell.Any) { return new Fly(self, Target.FromCell(self.World, cell)); }
-		public Activity VisualMove(Actor self, WPos fromPos, WPos toPos) { return Util.SequenceActivities(new CallFunc(() => SetVisualPosition(self, fromPos)), new Fly(self, Target.FromPos(toPos))); }
+		public Activity VisualMove(Actor self, WPos fromPos, WPos toPos)
+		{ return Util.SequenceActivities(new CallFunc(() => SetVisualPosition(self, fromPos)), new Fly(self, Target.FromPos(toPos))); }
```
Comment (Mailaender) [inline OpenRA.Mods.Common/Traits/Air/Plane.cs]: This looks a bit weird. Maybe split it into several lines with
```csharp
{
	return ...
}
```

Comment (RoosterDragon) [inline OpenRA.Mods.Common/Traits/Air/Plane.cs]: Done.

**Comment by RoosterDragon** [comment]
Seems like we have a rough consensus for "180 now, maybe lower later".

**Comment by Mailaender** [comment]
Most of the code changes here make things more readable. I personally usually always ignore those line lengths standards that are common in the industry, because it is easier for me to see the whole if statement in one line and I have a wide screen monitor, but I have to adapt sometime anyway. I see the benefits for the community especially in code review and diffs. Thanks for this. After merging, every pull request with lengthy lines should rebase first.

**Comment by penev92** [comment]
I'm on a 1400x900 monitor and I have a couple of IDE add-ons that make my text area even smaller, so this is a good start :)

**Pull request merged**